docs/_tutorials/bert-pretraining.md

```diff
@@ -0,0 +1,319 @@
+4. `fp16`: enable FP16 mixed precision training with an initial loss scale factor 2^16.
+
+That's it! That's all you need do in order to use DeepSpeed in terms of
+modifications. We have included a modified `train.py` file called
+`DeepSpeedExamples/bing_bert/deepspeed_train.py` with all of the changes
+applied.
+
+
+### Start Training
+An example of launching `deepspeed_train.py` on four nodes with four GPUs each would be:
+```bash
+deepspeed --num_nodes 4  \
+    deepspeed_train.py \
+    --deepspeed \
+    --deepspeed_config  deepspeed_bsz4096_adam_config.json
+    --cf /path-to-deepspeed/examples/tests/bing_bert/bert_large_adam_seq128.json \
+    --train_batch_size 4096  \
+    --max_seq_length 128 \
+    --gradient_accumulation_steps 4 \
+    --max_grad_norm 1.0 \
+    --fp16 \
+    --loss_scale 0 \
+    --delay_allreduce \
+    --max_steps 32 \
+    --print_steps 1 \
+    --output_dir <output_directory>
+```
+See the [Getting Started](/getting-started/) guide for more information on
+launching DeepSpeed.
+
+------
+
+## Reproducing BERT Training Results with DeepSpeed
+
+Our BERT training result is competitive across the industry in terms of
+achieving F1 score of 90.5 or better on the SQUAD 1.1 dev set:
+
+- Comparing with the original BERT training time from Google, it took them
+about 96 hours to reach parity on 64 TPU2 chips, while it took us 14 hours on
+4 DGX-2 nodes of 64 V100 GPUs.
+- On 256 GPUs, it took us 3.7 hours, faster than state-of-art result (3.9
+hours) from Nvidia using their superpod on the same number of GPUs
+([link](https://devblogs.nvidia.com/training-bert-with-gpus/)).
+
+![BERT Training Time](/assets/images/bert-large-training-time.png){: .align-center}
+
+Our configuration for the BERT training result above can be reproduced with
+the scripts/json configs in our DeepSpeed repo. Below is a table containing a
+summary of the configurations. Specifically see the
+`ds_train_bert_bsz16k_seq128.sh` and `ds_train_bert_bsz16k_seq512.sh` scripts
+for more details in
+[DeepSpeedExamples](https://github.com/microsoft/DeepSpeedExamples/tree/master/bing_bert).
+
+
+| Parameters               | 128 Sequence              | 512 Sequence              |
+| ------------------------ | ------------------------- | ------------------------- |
+| Total batch size         | 16K                       | 16K                       |
+| Train micro batch size per gpu | 64                  | 8                         |
+| Optimizer                | Lamb                      | Lamb                      |
+| Learning rate            | 4e-3                      | 1e-3                      |
+| Min Lamb coefficient     | 0.08                      | 0.08                      |
+| Max Lamb coefficient     | 0.5                       | 0.5                       |
+| Learning rate scheduler  | `warmup_linear_decay_exp` | `warmup_linear_decay_exp` |
+| Warmup ratio             | 0.02                      | 0.01                      |
```

Comment:
In the code itself the 'warmup ratio' parameter is actually set with 'warmup_proportion'

Reply:
Fixed